src/source_info/lines.ml

```diff
@@ -0,0 +1,21 @@
+let lines_locs src =
+  let lines = String.split_on_char '\n' src in
+  let is_line_directive line =
+    if String.length line > 0 then String.get line 0 = '#' else false
+  in
+  let _, poses, _ =
+    List.fold_left
+      (fun (i, poses, count) line ->
+        let l = String.length line in
+        let new_i, new_pos =
+          (* Ignore line directives *)
+          if is_line_directive line then (i, poses)
```

Comment:
What is this doing ?
I think line directives are already counted in the locations. There's a directive for defining the line number (starting on the next line), perhaps that's what you meant to handle ?

```
# 31 "src/loader/cmi.ml"
```

Reply:
Yes, this code was ignoring line directives only to have a working demo. I included the code in this branch by mistake, sorry about that. Line directives need to be handled correctly (this is #15). @EmileTrotignon and I are working on this...